src/ray/core_worker/core_worker.cc

```diff
@@ -750,9 +750,8 @@ CoreWorker::CoreWorker(CoreWorkerOptions options, const WorkerID &worker_id)
   // Tell the raylet the port that we are listening on.
   // NOTE: This also marks the worker as available in Raylet. We do this at the
   // very end in case there is a problem during construction.
-  if (options.connect_on_start) {
-    ConnectToRayletInternal();
-  }
+  ConnectToRayletInternal();
```

Comment:
We should move this to the very end of the constructor.

Reply:
> This will match the current behavior.

I didn't change any code position..

Reply:
today's behavior: if driver, connect here. if worker, connect right after ctor.

Reply:
My understanding is that this PR unifies driver and worker connect and I think following the worker connect position is better: i.e. connect the very end of the constructor.

Reply:
I moved to the end of the ctor.